File: src/transformers/features/misc/MiscCreateAssertPruneTransformer.ts
import { MiscAssertPruneProgrammer } from "../../../programmers/misc/MiscAssertPruneProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace MiscCreateAssertPruneTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "misc.createAssertPrune",
      write: MiscAssertPruneProgrammer.write,
    });
}

File: packages/transform/src/features/misc/MiscCreateAssertPruneTransformer.go
package misc

import (
	coremisc "github.com/samchon/typia/packages/core/src/programmers/misc"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var MiscCreateAssertPruneTransformer = miscCreateAssertPruneTransformerNamespace{}

type miscCreateAssertPruneTransformerNamespace struct{}

func (miscCreateAssertPruneTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coremisc.MiscPruneProgrammer.Write(call.TypeText)
}
